File: get_2025_draft.py
import requests
import csv
import json

# Your league ID from leagueInfo.js
LEAGUE_ID = "1185988161254850560"

def get_2025_draft_picks():
    try:
        print('Fetching 2025 draft data...')
        
        # First, get the league data to find the draft ID
        league_response = requests.get(f'https://api.sleeper.app/v1/league/{LEAGUE_ID}')
        league_data = league_response.json()
        
        if not league_data.get('draft_id'):
            print('No draft found for this league')
            return
        
        draft_id = league_data['draft_id']
        print(f'Found draft ID: {draft_id}')
        
        # Get the draft details
        draft_response = requests.get(f'https://api.sleeper.app/v1/draft/{draft_id}')
        draft_data = draft_response.json()
        
        # Get the draft picks
        picks_response = requests.get(f'https://api.sleeper.app/v1/draft/{draft_id}/picks')
        picks_data = picks_response.json()
        
        # Get player data for position information
        players_response = requests.get('https://api.sleeper.app/v1/players/nfl')
        players_data = players_response.json()
        
        print(f'Found {len(picks_data)} draft picks')
        
        # Process the picks
        processed_picks = []
        for pick in picks_data:
            player = players_data.get(pick.get('player_id', ''))
            pick_number = pick.get('pick_no', 0)
            round_num = pick.get('round', 0)
            position = player.get('position', 'Unknown') if player else 'Unknown'
            player_name = f"{player.get('first_name', '')} {player.get('last_name', '')}" if player else 'Unknown Player'
            team = player.get('team', 'FA') if player else 'FA'
            
            processed_picks.append({
                'pick_number': pick_number,
                'round': round_num,
                'position': position,
                'player_name': player_name.strip(),
                'team': team,
                'manager': pick.get('picked_by', 'Unknown')
            })
        
        # Sort by pick number
        processed_picks.sort(key=lambda x: x['pick_number'])
        
        # Write to CSV file
        with open('2025_draft_picks.csv', 'w', newline='', encoding='utf-8') as csvfile:
            fieldnames = ['pick_number', 'round', 'position', 'player_name', 'team', 'manager']
            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
            
            writer.writeheader()
            for pick in processed_picks:
                writer.writerow(pick)
        
        print('\n=== 2025 DRAFT PICKS ===')
        print('Pick # | Round | Pos | Player Name | Team | Manager')
        print('--------|-------|-----|-------------|------|---------')
        
        for pick in processed_picks:
            print(f"{pick['pick_number']:6} | {pick['round']:5} | {pick['position']:3} | {pick['player_name']:20} | {pick['team']:3} | {pick['manager']}")
        
        print(f'\nData exported to "2025_draft_picks.csv"')
        print(f'Total picks: {len(processed_picks)}')
        
        # Summary by position
        position_counts = {}
        for pick in processed_picks:
            pos = pick['position']
            position_counts[pos] = position_counts.get(pos, 0) + 1
        
        print('\n=== POSITION BREAKDOWN ===')
        for pos, count in sorted(position_counts.items()):
            print(f'{pos}: {count} picks')
        
        # Summary by round
        round_counts = {}
        for pick in processed_picks:
            round_num = pick['round']
            round_counts[round_num] = round_counts.get(round_num, 0) + 1
        
        print('\n=== ROUND BREAKDOWN ===')
        for round_num in sorted(round_counts.keys()):
            print(f'Round {round_num}: {round_counts[round_num]} picks')
        
    except Exception as error:
        print(f'Error fetching draft data: {error}')

if __name__ == "__main__":
    get_2025_draft_picks()

File: get_2025_draft.js
const fs = require('fs');

// Your league ID from leagueInfo.js
const LEAGUE_ID = "1185988161254850560";

async function get2025DraftPicks() {
    try {
        console.log('Fetching 2025 draft data...');
        
        // First, get the league data to find the draft ID
        const leagueResponse = await fetch(`https://api.sleeper.app/v1/league/${LEAGUE_ID}`);
        const leagueData = await leagueResponse.json();
        
        if (!leagueData.draft_id) {
            console.log('No draft found for this league');
            return;
        }
        
        console.log(`Found draft ID: ${leagueData.draft_id}`);
        
        // Get the draft details
        const draftResponse = await fetch(`https://api.sleeper.app/v1/draft/${leagueData.draft_id}`);
        const draftData = await draftResponse.json();
        
        // Get the draft picks
        const picksResponse = await fetch(`https://api.sleeper.app/v1/draft/${leagueData.draft_id}/picks`);
        const picksData = await picksResponse.json();
        
        // Get player data for position information
        const playersResponse = await fetch('https://api.sleeper.app/v1/players/nfl');
        const playersData = await playersResponse.json();
        
        console.log(`Found ${picksData.length} draft picks`);
        
        // Process the picks
        const processedPicks = picksData.map(pick => {
            const player = playersData[pick.player_id];
            const pickNumber = pick.pick_no;
            const round = pick.round;
            const position = player ? player.position : 'Unknown';
            const playerName = player ? `${player.first_name} ${player.last_name}` : 'Unknown Player';
            const team = player ? player.team : 'FA';
            
            return {
                pickNumber,
                round,
                position,
                playerName,
                team,
                manager: pick.picked_by || 'Unknown'
            };
        });
        
        // Sort by pick number
        processedPicks.sort((a, b) => a.pickNumber - b.pickNumber);
        
        // Create CSV content for Excel
        let csvContent = 'Pick Number,Round,Position,Player Name,Team,Manager\n';
        
        processedPicks.forEach(pick => {
            csvContent += `${pick.pickNumber},${pick.round},${pick.position},${pick.playerName},${pick.team},${pick.manager}\n`;
        });
        
        // Write to file
        fs.writeFileSync('2025_draft_picks.csv', csvContent);
        
        console.log('\n=== 2025 DRAFT PICKS ===');
        console.log('Pick # | Round | Pos | Player Name | Team | Manager');
        console.log('--------|-------|-----|-------------|------|---------');
        
        processedPicks.forEach(pick => {
            console.log(`${pick.pickNumber.toString().padStart(6)} | ${pick.round.toString().padStart(5)} | ${pick.position.padStart(3)} | ${pick.playerName.padEnd(20)} | ${pick.team.padStart(3)} | ${pick.manager}`);
        });
        
        console.log(`\nData exported to '2025_draft_picks.csv'`);
        console.log(`Total picks: ${processedPicks.length}`);
        
        // Summary by position
        const positionCounts = {};
        processedPicks.forEach(pick => {
            positionCounts[pick.position] = (positionCounts[pick.position] || 0) + 1;
        });
        
        console.log('\n=== POSITION BREAKDOWN ===');
        Object.entries(positionCounts).forEach(([pos, count]) => {
            console.log(`${pos}: ${count} picks`);
        });
        
    } catch (error) {
        console.error('Error fetching draft data:', error);
    }
}

// Run the function
get2025DraftPicks();
